tutorial1.md

```diff
@@ -0,0 +1,299 @@
+Mar 23 01:47:27 k8s-tutorial kubelet[12011]: F0323 01:47:27.767706   12011 server.go:261] failed to run Kubelet: unable to load bootstrap kubeconfig: stat /etc/kubernetes/bootstrap-kubelet.conf: no such file or directory
+Mar 23 01:47:27 k8s-tutorial systemd[1]: kubelet.service: Main process exited, code=exited, status=255/n/a
+Mar 23 01:47:27 k8s-tutorial systemd[1]: kubelet.service: Failed with result 'exit-code'.
+```
+
+Great, a new error message! let's tackle this one.
+Looking at the [kubelet docs][kubelet], we can see that `--bootstrap-kubeconfig` is used for a kubeconfig we'll use to retrieve our real kubeconfig.
+Since we don't have a kubelet yet, we can just comment this out.
+
+[kubelet]: https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/
+
+```shell
+sudoedit /etc/systemd/system/kubelet.service.d/10-kubeadm.conf
+```
+
+Comment out the line that starts with `Environment="KUBELET_KUBECONFIG_ARGS=`:
+
+```shell
+Environment="KUBELET_KUBECONFIG_ARGS=
+```
+
+## Disable authentication
+
+We've got a new error to deal with:
+
+```console
+$ sudo journalctl --lines 5 --unit kubelet
+-- Logs begin at Fri 2019-03-22 19:46:26 UTC, end at Sat 2019-03-23 02:22:43 UTC. --
+Mar 23 02:22:39 k8s-tutorial kubelet[20264]: I0323 02:22:39.259702   20264 plugins.go:103] No cloud provider specified.
+Mar 23 02:22:39 k8s-tutorial kubelet[20264]: W0323 02:22:39.259867   20264 server.go:552] standalone mode, no API client
+Mar 23 02:22:39 k8s-tutorial kubelet[20264]: F0323 02:22:39.260025   20264 server.go:261] failed to run Kubelet: no client provided, cannot use webhook authentication
+Mar 23 02:22:39 k8s-tutorial systemd[1]: kubelet.service: Main process exited, code=exited, status=255/n/a
+Mar 23 02:22:39 k8s-tutorial systemd[1]: kubelet.service: Failed with result 'exit-code'.
+```
+
+We don't need authentication! Let's turn it off.
+Peeking through the [kubelet configuration][kubeletdoc], we can see that there's an object that [controls authentication][kubeletauth]. This is the default:
+
+[kubeletdoc]: https://godoc.org/k8s.io/kubelet/config/v1beta1#KubeletConfiguration
+[kubeletauth]: https://godoc.org/k8s.io/kubelet/config/v1beta1#KubeletAuthentication
+
+```yaml
+anonymous:
+  enabled: false
+webhook:
+  enabled: true
+  cacheTTL: "2m"
+```
+
+Let's override that:
+
+```yaml
+kind: KubeletConfiguration
+apiVersion: kubelet.config.k8s.io/v1beta1
+authentication:
+  webhook:
+    enabled: false
+  anonymous:
+    enabled: true
+authorization:
+  mode: AlwaysAllow
+```
+
+authentication and authorization are different.
+authentication answers "who are you," and authorization answers "are you allowed."
+To get the server to work in standalone mode, we have to give answers to both those questions.
+
+If all's gone well, you should see some log lines that look like this:
+
+```console
+$ sudo journalctl --lines 10 --unit kubelet
```

Comment:
Active: active (running) since Sat 2019-03-23 19:33:46 EDT; 1min 11s ago
as well in `sudo systemctl status kubelet`